syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

message OpenSearch {

  message QueryLogs {
    google.protobuf.StringValue index = 1;
    google.protobuf.StringValue query_dsl = 2;
    google.protobuf.UInt64Value limit = 3;
    google.protobuf.UInt64Value offset = 4;
    google.protobuf.StringValue sort_desc = 5;
    google.protobuf.StringValue timestamp_field = 6;
  }

  message DeleteIndex {
    google.protobuf.StringValue index = 1;
  }

  message GetNodeStats {}

  message GetIndexStats {}

  enum TaskType {
    UNKNOWN = 0;
    QUERY_LOGS = 1;
    DELETE_INDEX = 2;
    GET_NODE_STATS = 3;
    GET_INDEX_STATS = 4;
  }

  TaskType type = 1;
  oneof task {
    QueryLogs query_logs = 101;
    DeleteIndex delete_index = 102;
    GetNodeStats get_node_stats = 103;
    GetIndexStats get_index_stats = 104;
  }
}